tests/adam_test.cpp

```diff
@@ -460,3 +460,20 @@ TEST_CASE("QHAdamLogisticRegressionTest", "[AdamTest]")
       coordinates);
   REQUIRE(testAcc == Approx(100.0).epsilon(0.006)); // 0.6% error tolerance.
 }
+
+/**
+ * Test the Adam optimizer on the Ackley function.
+ * This is to test the Ackley function and not Adam.
+ * This test will be removed later.
+ */
+TEST_CASE("AdamAckleyFunctionTest", "[AdamTest]")
```

Comment:
The test looks good to me, also I agree we can remove the test later once PSO is merged.